src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs

```diff
@@ -1482,6 +1482,11 @@ public new CSharpCommandLineArguments Parse(IEnumerable<string> args, string? ba
                                                  new CSharpRequiredLanguageVersion(MessageID.IDS_FeatureNullableReferenceTypes.RequiredVersion())), Location.None));
             }
 
+            // Sort so that more specific keys precede less specific.
+            // When mapping a path we find the first key in the array that is a prefix of the path.
+            // If multiple keys are prefixes of the path we want to use the longest (more specific) one for the mapping.
+            pathMap = pathMap.Sort((x, y) => -x.Key.Length.CompareTo(y.Key.Length));
```

Comment:
Why are we only doing this for C#? VB also supports path mapping and believe we'd want to be consistent here.

Reply:
Oh, good catch. Forgot about VB. Will add this to VB as well.